controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -439,26 +442,88 @@ func (r *ROSAControlPlaneReconciler) reconcileClusterVersion(rosaScope *scope.RO
+		regConfig.AllowedRegistriesForImport = rosaControlPlane.Spec.ClusterRegistryConfig.AllowedRegistriesForImport
 
-	// if this fails, the provided role is likely invalid or it doesn't have the required permissions.
-	if err := ocmClient.UpdateCluster(cluster.ID(), creator, ocmClusterSpec); err != nil {
-		conditions.MarkFalse(rosaScope.ControlPlane,
-			rosacontrolplanev1.ROSAControlPlaneValidCondition,
-			rosacontrolplanev1.ROSAControlPlaneInvalidConfigurationReason,
-			clusterv1.ConditionSeverityError,
-			err.Error())
-		return err
+		if rosaControlPlane.Spec.ClusterRegistryConfig.RegistrySources != nil {
+			regConfig.RegistrySources.AllowedRegistries = rosaControlPlane.Spec.ClusterRegistryConfig.RegistrySources.AllowedRegistries
+			regConfig.RegistrySources.BlockedRegistries = rosaControlPlane.Spec.ClusterRegistryConfig.RegistrySources.BlockedRegistries
+			regConfig.RegistrySources.InsecureRegistries = rosaControlPlane.Spec.ClusterRegistryConfig.RegistrySources.InsecureRegistries
+		}
+	}
+	if !reflect.DeepEqual(regConfig.AdditionalTrustedCa, cluster.RegistryConfig().AdditionalTrustedCa()) {
+		ocmClusterSpec.AdditionalTrustedCa = regConfig.AdditionalTrustedCa
+		updated = true
+	}
+	if !reflect.DeepEqual(regConfig.RegistrySources.AllowedRegistries, cluster.RegistryConfig().RegistrySources().AllowedRegistries()) {
+		ocmClusterSpec.AllowedRegistries = regConfig.RegistrySources.AllowedRegistries
+		updated = true
+	}
+	if !reflect.DeepEqual(regConfig.RegistrySources.BlockedRegistries, cluster.RegistryConfig().RegistrySources().BlockedRegistries()) {
+		ocmClusterSpec.BlockedRegistries = regConfig.RegistrySources.BlockedRegistries
+		updated = true
+	}
+	if !reflect.DeepEqual(regConfig.RegistrySources.InsecureRegistries, cluster.RegistryConfig().RegistrySources().InsecureRegistries()) {
+		ocmClusterSpec.InsecureRegistries = regConfig.RegistrySources.InsecureRegistries
+		updated = true
 	}
 
-	return nil
+	var newAllowedRegisters, oldAllowedRegisters []string
+	if len(regConfig.AllowedRegistriesForImport) > 0 {
+		for id := range regConfig.AllowedRegistriesForImport {
+			newAllowedRegisters = append(newAllowedRegisters, regConfig.AllowedRegistriesForImport[id].DomainName+":"+
+				strconv.FormatBool(regConfig.AllowedRegistriesForImport[id].Insecure))
+		}
+	}
+	if len(cluster.RegistryConfig().AllowedRegistriesForImport()) > 0 {
+		for id := range cluster.RegistryConfig().AllowedRegistriesForImport() {
+			oldAllowedRegisters = append(oldAllowedRegisters, cluster.RegistryConfig().AllowedRegistriesForImport()[id].DomainName()+":"+
+				strconv.FormatBool(cluster.RegistryConfig().AllowedRegistriesForImport()[id].Insecure()))
+		}
+	}
+	if !reflect.DeepEqual(newAllowedRegisters, oldAllowedRegisters) {
+		ocmClusterSpec.AllowedRegistriesForImport = strings.Join(newAllowedRegisters, ",")
+		updated = true
+	}
+
+	// TODO: check for cluster AutoScale changes
```

Comment:
Are you planning to do this in this PR, or a follow up?

Reply:
it should be in another PR , suppose to be a bug.